content/multi-level-enums.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: sip
+permalink: /sips/:title.html
+stage: implementation
+status: waiting-for-implementation
+presip-thread: https://contributors.scala-lang.org/t/pre-sip-foo-bar/9999
+title: SIP-NN - Multi-Level Enums
+---
+
+# SIP: Multi-Level Enums
+
+**By: Raphael Bosshard**
+
+## History
+
+| Date       | Version            |
+| ---------- | ------------------ |
+| 2025-09-15 | Initial Draft      |
+
+
+
+## Summary
+
+This proposal adds minimal syntax to allow enums with nested enumerations, maintaining full exhaustivity checking while keeping syntax clean and intuitive.
+
+## Motivation
+
+Scala 3 introduced `enum` as a concise and type-safe way to define algebraic data types. However, it currently supports only flat enums. Many real-world use cases, such as domain modeling or UI state machines, naturally require **hierarchical enums**, where cases are grouped into logical families or categories.
+
+Consider modeling animals:
+
+```scala
+enum Animal:
+  case Dog, Cat, Sparrow, Penguin
+```
+
+This flat structure works, but grouping `Dog` and `Cat` under `Mammal`, and `Sparrow` and `Penguin` under `Bird` is more expressive and enables cleaner abstraction.
+
+
+## Proposed solution
+
+### High-level overview
+
+Allow `enum` cases to contain **nested `enum` definitions**, using a consistent indentation-based syntax.
+
+```scala
+enum Animal:
+  case enum Mammal:
+    case Dog, Cat
+  case enum Bird:
+    case Sparrow, Pinguin
```

Comment:
also good to demonstrate how parameters work (e.g. `case enum Bird(val call: String):`) and optional extends clauses